### DISCUSSION TITLE
Team colour in .results["TeamColor"] vs. .team_color() different?

### DISCUSSION THREAD

**Discussion by AreaXimus1** [theOehrly/Fast-F1#527] [Q&A]
I've done a fastest lap comparison for Day 2 of testing and Perez is yellow? This is using the team colour gathering example from the [documentation](https://docs.fastf1.dev/examples_gallery/plot_qualifying_results.html). The rest of the teams look very accurate, but Red Bull (or at least PER) doesn't seem accurate. 
Meanwhile the hex code in .results["TeamColor"] is blue. I'm not exactly sure why.

![image](https://github.com/theOehrly/Fast-F1/assets/154688263/fbf530c5-047d-46ef-9429-3e38d4c8832b)
![image](https://github.com/theOehrly/Fast-F1/assets/154688263/bf343898-2437-4382-bf4e-6bb595307899)

Update: I've had a look at all other teams and it seems to be a feature not a bug. This table compares session.result["TeamColor"] and fastf1.plotting.teamcolor() and they are consistently different.
![image](https://github.com/theOehrly/Fast-F1/assets/154688263/92bfb153-1dbd-42fe-b12d-5f71ba416f3a)

**Comment by Casper-Guo** [comment]
Can you post the source code you used please?

**Reply by AreaXimus1** [reply]
My code: 


`team_colours_day2 = list()`
`for index, lap in fastest_laps_day2.iterlaps():`
`    colour = fastf1.plotting.team_color(lap["Team"])`
`    team_colours_day2.append(colour)`

I've now update my original post to include a link to the documentation I used as an example.

Looking at the comparison between day2.results["TeamColor"] and the results that this fastf1.plotting.team_color() generates, they're not really the same for any team.

**Comment by Casper-Guo** [comment]
Yea the color palette is changed in version 3.3.0 for the 2024 season. Not sure why the maintainer chooses the current set of colors. The process is very much ad-hoc

**Reply by AreaXimus1** [reply]
Yeah I can understand that differentiation is important, but a yellow Red Bull is nothing we've seen in any official F1 media.

**Comment by theOehrly** [comment]
The team color in the results is what the API returns. FastF1s plotting colors are independent from that.
RedBull being yellow is an intentional new change, else we have six drivers in various shades of blue that you can barely tell apart. And RedBull is using yellow as an accent color in their branding and on their car, that's were the color comes from.

**Reply by AreaXimus1** [reply]
Thanks for your quick response! I can definitely understand why you'd want the colour to be yellow -- the teams haven't exactly made it easy for us! 
I can't say I like it, but the joy of this package/matplotlib in general is that it's pretty easy to change it about. 

This is my first day of using it, but I really appreciate what I'm able to do with this!

**Reply by theOehrly** [reply]
My reasons for this were basically the following:
You were already not able to look at a plot and say "ohh blue, that's Red Bull" because there were three to four blue teams. Therefore, making Red Bull yellow is not really loosing instant recognizability. Yellow is one of their accent colors. Alpine is already pink now, but arguably blue would be their primary color too, so an accent color is used there too.
On the plus side, yellow is much more distinguishable from the other colors and massively improves the readability of the plot. 

I'm open for discussion here. I know that people aren't used to this color. Subjectively and in the context of creating data visualization, I think it is the better color.
